### ISSUE TITLE
Morph feature requests

### ISSUE THREAD

**Issue by vcvrackideas** [23volts/23volts-vcv#7]
Hi again Remi!
- Do you plan a bigger version of the Morph? 4 snapshots are cool but sometimes arrangement requires more, like 8 ones. 
- It would be also good to have a jump in assign mode as I described in Multimap module to increase assigning process a twice ;-)
- Some kind of switch (if you have a room for this of course) between A-B-C-D would be handy for inputting snapshots values. Let's say if I didn't notice a highlight on specific letter (in creative process, haha) all of my set snapshot values get lost in the process...
- And last but not least) XY inputs are ok but I guess for live performance (or for recording an automation with Stoermelder's Remove module) I think the best option would be to have specific X and Y knobs for midi mapping (again if you have a room for this). If you don't have a room I can take a knob with cv output but again it's not so fast as can be with the knobs.

**Comment by RemiCollin** [comment]
Hey !

> * Do you plan a bigger version of the Morph? 4 snapshots are cool but sometimes arrangement requires more, like 8 ones.

Yes, but not in the near future, as it needs code refactoring a little bit and I'm not 100% happy about the module's workflow yet, once this is settled I'll plan a bigger version. 



> * It would be also good to have a jump in assign mode as I described in Multimap module to increase assigning process a twice ;-)

It uses the same code from Multimap, so once it will be implemented it will be available in morph as well. 



> * Some kind of switch (if you have a room for this of course) between A-B-C-D would be handy for inputting snapshots values. Let's say if I didn't notice a highlight on specific letter (in creative process, haha) all of my set snapshot values get lost in the process...

Not sure how i'm gonna approach this, these are the options I'm considering  : 

1) it needs to be obvious to the user when the snapshot is in "Record" mode and when it's not. Maybe have the snapshot's letter colored/blinking.. ? 

2) Locking the knobs in between snapshots is maybe what's make more sense, but i'm not sure if it would feel weird to the user

> * And last but not least) XY inputs are ok but I guess for live performance (or for recording an automation with Stoermelder's Remove module) I think the best option would be to have specific X and Y knobs for midi mapping (again if you have a room for this). If you don't have a room I can take a knob with cv output but again it's not so fast as can be with the knobs.

Direct Midi mapping the X and Y axis is already in the Roadmap ;)

**Comment by vcvrackideas** [comment]
> Not sure how i'm gonna approach this, these are the options I'm considering :
> 
> 1. it needs to be obvious to the user when the snapshot is in "Record" mode and when it's not. Maybe have the snapshot's letter colored/blinking.. ?
> 2. Locking the knobs in between snapshots is maybe what's make more sense, but i'm not sure if it would feel weird to the user

yeah I think the first variant is what we need in this module.

**Comment by vcvrackideas** [comment]
nice to get the update! It works with separate knobs on my touchosc pretty well. But I can't map it with XY pad on touchosc, learn just switching to the next one and vice versa. I tested it with midi cc and it seems touchosc corresponds CC well. I can map it through the multimap for example but morph doesn't allow to do internal mapping due to have midi device inside)
edit: and yeah, chain assignment now works really faster than ever! thanks very much for this

**Comment by RemiCollin** [comment]
I had the same issue with TouchOSC some time ago as it sends both CC at the same time, it was more or less a matter of luck to map the correct axis. I workarounded it by creating a specific page with 2 faders with the same CC/Channel just for assignment.

**Comment by vcvrackideas** [comment]
So I did some tests to make myself able to ask something about improving the module.
- I noticed a some lag while tweaking already assigned knobs on Multimap itself when mapping to MindMeldJr mixer (to faders)
- Still thinking about ABCD point. I think if you can increase the room of the module or add some expanders there would be a nice results if you can add here a direct buttons to A B C and D to allow user edit values in the point. And so these knobs would have ability to map them from controller so the process can be faster in that way.
- So if we would have a room or expander I prefer to have a knobs that would control the GREY square instead of touching the mouse. If we now have ability to use XY cv inputs or internal mapping we  can just adjust only BLUE square (modulation one) so the starting point not A but C. A bit confusing honestly.

**Comment by RemiCollin** [comment]
> * noticed a some lag while tweaking already assigned knobs on Multimap itself when mapping to MindMeldJr mixer (to faders)

I did a quick test but it doesn't seem to lag on my side. I know that mixmeister faders can behave a little strangely (seems to have some sort of smoothing on them). Does it happens on other modules ?



> * Still thinking about ABCD point. I think if you can increase the room of the module or add some expanders there would be a nice results if you can add here a direct buttons to A B C and D to allow user edit values in the point. And so these knobs would have ability to map them from controller so the process can be faster in that way.

I'll think about it. I probably can make room for 4 buttons by tweaking the layout a little bit. 

> * So if we would have a room or expander I prefer to have a knobs that would control the GREY square instead of touching the mouse. If we now have ability to use XY cv inputs or internal mapping we  can just adjust only BLUE square (modulation one) so the starting point not A but C. A bit confusing honestly.

Do you mean when using knobs as inputs for XY Cv inputs ? Not sure I get it

**Comment by vcvrackideas** [comment]
> I did a quick test but it doesn't seem to lag on my side. I know that mixmeister faders can behave a little strangely (seems to have some sort of smoothing on them). Does it happens on other modules ?

Yep there is an issue only with MindMeld because it works ok with bogaudio faders. Sorry. The knob just stuck in the zero region, in the middle, pretty strange. But it still works ok with midi controller, pretty pretty strange.



> Do you mean when using knobs as inputs for XY Cv inputs ? Not sure I get it

yeah, direct knobs, on for X and another for Y GREY (main, non modulation one) square instead of touching mouse.

**Comment by RemiCollin** [comment]
> Yep there is an issue only with MindMeld because it works ok with bogaudio faders. Sorry. The knob just stuck in the zero region, in the middle, pretty strange. But it still works ok with midi controller, pretty pretty strange.

Probably worth a shot to ask Marc Boulé about this.



> yeah, direct knobs, on for X and another for Y GREY (main, non modulation one) square instead of touching mouse.

Ok. Will probably wait for Rack V2 API for this one, as it will allows me to implement MIDI learn with right click, so the layout will be cleaner.

**Comment by vcvrackideas** [comment]
Thanks! Also what do you think on adding in the context menu such commands as "Copy current state of knobs to A" and same command for B, C and D and also "copy current state of knobs to all presets"? Just to make precise changes while turning knobs and then morphing without rolloff to default 12 o'clock values

**Comment by RemiCollin** [comment]
Yes should be no big deal.
